### ISSUE TITLE
wrong result of mask-rcnn in c++

### ISSUE THREAD

**Issue by xuexl** [facebookresearch/detectron2#3143]
I train a model in detectron2, using mask-rcnn . And in my dataset, there is only one class, that is water. Then export model by "export_model.py" to "libtorch", and i run the model , and i find that the result is not same the result in the detectron2.

/********************************************************/
bbox: CPUFloatType [8, 4]
483.2622  499.7980  525.8268  572.2922
 498.9185  497.9980  542.1021  575.2800
 459.5051  489.5257  501.0689  576.5564
   1.8221  473.4080  352.0740  571.5385
   0.6046  484.8984  165.6757  547.0325
  10.9289  519.1484  174.4597  542.2369
 945.5907  568.6548  954.6726  591.4166
 443.3625  489.2993  484.0106  574.4414

scores: CPUFloatType [8]
0.01 *
 5.6323
 5.4494
 5.4176
 5.3469
 5.2617
 5.2389
 5.1474
 5.0809

pred_classes: CPULongType [8]
29
 29
 29
 29
 29
 29
 57
 29
pred_masks: CPUFloatType [8, 1, 28, 28]
Columns 1 to 9  0.4994  0.4996  0.4987  0.4990  0.4982  0.4987  0.4978  0.4984  0.4977
  0.4998  0.4993  0.4989  0.4984  0.4984  0.4979  0.4981  0.4975  0.4983
  0.4992  0.4995  0.4985  0.4991  0.4981  0.4987  0.4977  0.4984  0.4973
  0.4995  0.4993  0.4990  0.4987  0.4985  0.4983  0.4980  0.4978  0.4980
  0.5001  0.5005  0.4999  0.5005  0.4996  0.5002  0.4993  0.4999  0.4988
....
/********************************************************/

There is one class in my dataset, but the "pred_classes" is  '29' or '57'  , why ?
and  the size of  "pred_masks"  is [28, 28], and the data are "0.4998 ..." , that is strange,
i can not understand.

please help me, thanks a lot.

**Comment by xuexl** [comment]
The progress of my train like this :
/***************************************************/

cfg = get_cfg()
cfg.merge_from_file(model_zoo.get_config_file("COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml"))
cfg.DATASETS.TRAIN = ("water_train",)
cfg.DATASETS.TEST = ()
cfg.DATALOADER.NUM_WORKERS = 1
cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url("COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml")  
cfg.SOLVER.IMS_PER_BATCH = 1
cfg.SOLVER.BASE_LR = 0.00025  
cfg.SOLVER.MAX_ITER = 1000   
cfg.SOLVER.STEPS = []    
cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 128   
cfg.MODEL.ROI_HEADS.NUM_CLASSES =1  

os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)
trainer = DefaultTrainer(cfg) 
trainer.resume_or_load(resume=False)
trainer.train()

**Comment by xuexl** [comment]
This is the code of  export.
/******************************************************/

!python ./project/export_model.py --config-file ./project/configs/COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml \
    --output ./output --export-method tracing --format torchscript \
    MODEL.WEIGHTS ./project/weights/model_final.pth \
    MODEL.DEVICE cuda \
    --sample-image ./project/train/d1.jpg

**Comment by Denbergvanthijs** [comment]
Without postprocessing, output masks are of shape (n, 28, 28). You can resize them with the function detector_postprocess() from detectron2.modeling.postprocessing.py

**Comment by xuexl** [comment]
> 
> 
> Without postprocessing, output masks are of shape (n, 28, 28). You can resize them with the function detector_postprocess() from detectron2.modeling.postprocessing.py

/****/
Thanks a lot, Denbergvanthijs . 
Yes ,there are some methods to handle the output of model, but, when i export model  and develop in the libtorch , I get the wrong result. Maybe, the result need postprocessing , but i have no idea. 
So, What can i do in the postprocess ? Are there any examples about libtorch as the postprocess ?
